### PR TITLE
add json format support for traefik logs

### PR DESCRIPTION
**Pull request by marco-jantke**
This PR adds the possibility to use JSON format for the traefik logs. 

Also it aligns the configuration to the accessLogs and deprecates the `traefikLogsFile` option.

Sample log line:
```json
{"level":"info","msg":"Traefik version dev built on I don't remember exactly","time":"2017-09-05T02:34:52+02:00"}
```

This fixes https://github.com/containous/traefik/issues/780.

### CODE REVIEW

**Comment by marco-jantke** [comment]
@Juliens thanks for the review! I added one more commit in order to keep the colors when the logs are written to stdout :D :)

**Comment by marco-jantke** [comment]
@nmengin it's not about access logs :-) I could, of course, add an integration test for the traefik log itself, but I am not sure on what to assert.. What I could is pick some lines and verify that they are valid JSON, without looking into the contents. Do you think this would help?

**Comment by marco-jantke** [comment]
This is certainly a good point Timo. I tested with all the different constellations with CLI and toml configuration. The behaviour is consistent and as expected. Given the new configuration `traefikLog.filePath` is specified, it will always be used. Only in case it is not specified the deprecated option will be used.

I now added a warning about the usage of the deprecated and documented at the deprecated option documentation, that the new option takes precedence.

To summarise shortly:

`traefik --traefiklogsfile=./deprecated_log --traefiklog --traefiklog.filepath=./new_log --traefiklog.format="json"` will write to `./new_log` in JSON format. 

`traefik --traefiklogsfile=./deprecated_log --traefiklog --traefiklog.format="json"` will write to `./deprecated_log` in JSON format.
